src/components/drawing/index.js

```diff
@@ -689,15 +689,23 @@ drawing.textPointStyle = function(s, trace, gd) {
         selectedTextColorFn = fns.selectedTextColorFn;
     }
 
+    var template = trace.texttemplate;
     s.each(function(d) {
         var p = d3.select(this);
-        var text = Lib.extractOption(d, trace, 'tx', 'text');
+        var text = Lib.extractOption(d, trace, template ? 'txt' : 'tx', template ? 'texttemplate' : 'text');
 
         if(!text && text !== 0) {
             p.remove();
             return;
         }
 
+        if(template) {
+            var extra = {};
+            if(d.tx) extra.text = d.tx;
+            if(d.data) extra.customdata = d.data;
+            text = Lib.texttemplateString(text, {}, gd._fullLayout._d3locale, d, trace._meta || {}, extra);
```

Comment:
Hmm, sorry for just noticing this now. You're using the fields directly off `calcdata[i][j]` which have gone through `ax.d2c` 

For example, off this branch:

```js
Plotly.newPlot(gd, [{
  type: 'bar',
  textposition: 'outside',
  x: ['2019-01-01', '2019-02-01'],
  y: [1, 2],
  hovertemplate: '%{x}',
  texttemplate: '%{x}'
}])
```

gives:

![Peek 2019-07-24 09-25](https://user-images.githubusercontent.com/6675409/61797234-f6750380-adf4-11e9-8d84-66b70fdc21db.gif)

Instead, we should be using the `d` (d for data) filled in `fullData` - for values w/o a corresponding `textinfo` flag.

Hover uses [`appendArrayPointValue`](https://github.com/plotly/plotly.js/blob/96fe2624a0df16b50ceb3061ff18d95b6a2120da/src/components/fx/helpers.js#L159-L184) - maybe it could be reused here when `texttemplate` is set.

Reply:
> Hover uses [`appendArrayPointValue`](https://github.com/plotly/plotly.js/blob/96fe2624a0df16b50ceb3061ff18d95b6a2120da/src/components/fx/helpers.js#L159-L184) - maybe it could be reused here when `texttemplate` is set.

Using `appendArrayPointValue` improves things a lot: we get `customdata`, `lat`, `lon`, `location` and other variables available in the `texttemplate` without additional code. For the specific mock you provided, I get:
![newplot - 2019-07-24T125556 290](https://user-images.githubusercontent.com/301546/61812867-6940a780-ae12-11e9-8d57-d0f3d93ab227.png)
which is already a nice improvement. Should I aim to get the exact same formatting as in `hovertemplate`?

Reply:
Very good question. I can upsides to showing `Jan 1, 2019` and to showing `2019-01-01`.

Could be a good thing to discuss in tomorrow's meeting.

Reply:
As discussed, we should add support formatting dates using `d3-time-format` when a pipe operator is present ex.: `%{x|yyyy-mm-dd}`

Reply:
@etpinard Should I handle this in a separate PR?

Reply:
> Should I handle this in a separate PR?

I would prefer having just one texttemplate PR, but that's up to you.

Reply:
If I understand correctly, we have limited support for timezones at the moment (https://github.com/plotly/plotly.js/issues/1532, https://github.com/plotly/plotly.js/issues/1956) so is the following all we need https://github.com/plotly/plotly.js/compare/texttemplate...template-format-date ?

Reply:
Comment in -> https://github.com/plotly/plotly.js/commit/3a59d94281a19d90d220f5ae072a7dd6a5d3d7a7#r34758960

Reply:
Comment in -> https://github.com/plotly/plotly.js/commit/3a59d94281a19d90d220f5ae072a7dd6a5d3d7a7#r34758960